service/matching/matching_engine.go

```diff
@@ -163,6 +182,8 @@ type (
 		namespaceUpdateLockMapLock sync.Mutex
 		// Stores results of reachability queries to visibility
 		reachabilityCache reachabilityCache
+		// De-duping poll requests when starting/signaling deployment workflows
```

Comment:
This is added to adhere to the constraint:

`if same task queue and build ID appear in a poll request but with a different deployment name, the poll will be rejected`